### PR TITLE
Add a JIRA Issue ChangeSource

### PR DESCRIPTION
**Pull request by turran**
The idea is generate a change for every changed issue returned
by a JQL. With this patch, buildbot can trigger actions based
on a modification of an issue.

### CODE REVIEW

File: master/buildbot/changes/jirapoller.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
+from datetime import datetime, timedelta
+
+from twisted.internet import defer
+
+from buildbot.changes import base
+from buildbot.util.logger import Logger
+from buildbot.util.state import StateMixin
+from buildbot.util import httpclientservice
```
Comment (marecki315) [inline master/buildbot/changes/jirapoller.py:8]: httpclientservice imported but not used

File: master/buildbot/changes/jirapoller.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        super(JiraIssuePoller, self).__init__(user, token, domain, jql, **kwargs)
+
+    def checkConfig(self,
+                    user,
+                    token,
+                    domain,
+                    jql,
+                    **kwargs):
+        if not JIRA:
+            config.error("The python module 'jira' is needed to use a JIRA "
```
Comment (marecki315) [inline master/buildbot/changes/jirapoller.py:47]: looks like _config_ is not imported (missing _from buildbot import config_)

File: master/buildbot/changes/jirapoller.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+try:
+    from jira import JIRA
+except ImportError:
+    JIRA = None
+
+log = Logger()
+
+def dt_parse(t):
+    ret = datetime.strptime(t[0:16],'%Y-%m-%dT%H:%M')
+    if t[17]=='+':
```
Comment (marecki315) [inline master/buildbot/changes/jirapoller.py:21]: Code formatting in this file is inconsistent. Sometimes there are spaces before and after "=", sometimes there are not etc.

**Review by marecki315 — CHANGES_REQUESTED** [review]

File: master/buildbot/changes/jirapoller.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                     "category", "pollAtLaunch", "name")
+    db_class_name = 'JiraIssuePoller'
+
+    def __init__(self, user, token, domain, jql, **kwargs):
+        name = kwargs.get("name")
+        if not name:
+            kwargs["name"] = "JiraIssuePoller:" + domain + "/" + user + "/" + jql
+        super(JiraIssuePoller, self).__init__(user, token, domain, jql, **kwargs)
+
+    def checkConfig(self,
```
Comment (marecki315) [inline master/buildbot/changes/jirapoller.py:40]: user, token, domain, jql parameters are not used in this method

File: master/buildbot/changes/jirapoller.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    from jira import JIRA
+except ImportError:
+    JIRA = None
+
+log = Logger()
+
+def dt_parse(t):
+    ret = datetime.strptime(t[0:16],'%Y-%m-%dT%H:%M')
+    if t[17]=='+':
+       ret-=timedelta(hours=int(t[18:20]),minutes=int(t[20:]))
```
Comment (marecki315) [inline master/buildbot/changes/jirapoller.py:22]: wrong indent (3 spaces, not 4)

**Comment by turran** [comment]
I've addressed the comments on the fluendo-buildbot PR itself instead of having it here.

**Pull request closed without merging**